### PR TITLE
[X86][FP16] Fix crash issue when AVX512VL is not set

### PR DESCRIPTION
**Pull request by phoebewang**
Fixes problem reported on #116153.

### CODE REVIEW

File: llvm/test/CodeGen/X86/avx512-insert-extract.ll [hunk 1]
```diff
@@ -672,7 +672,7 @@ define <32 x i16> @insert_v32i16(<32 x i16> %x, i16 %y, ptr %ptr) nounwind {
 ; KNL-NEXT:    vinserti32x4 $0, %xmm1, %zmm0, %zmm1
 ; KNL-NEXT:    vmovd %edi, %xmm0
 ; KNL-NEXT:    vpbroadcastw %xmm0, %ymm0
-; KNL-NEXT:    vpternlogq $216, {{\.?LCPI[0-9]+_[0-9]+}}(%rip), %zmm1, %zmm0
+; KNL-NEXT:    vpternlogq {{.*#+}} zmm0 = zmm0 ^ (mem & (zmm0 ^ zmm1))
 ; KNL-NEXT:    retq
```
Comment (RKSimon) [inline llvm/test/CodeGen/X86/avx512-insert-extract.ll:676]: Please can you pre commit the ternlog changes? It's proving really messy having these in PRs ......

Comment (phoebewang) [reply llvm/test/CodeGen/X86/avx512-insert-extract.ll]: Done.

File: llvm/test/CodeGen/X86/fminimum-fmaximum.ll [hunk 2]
```diff
@@ -1641,26 +1641,188 @@ define <4 x half> @test_fmaximum_v4f16(<4 x half> %x, <4 x half> %y) {
...
-; AVX512-NEXT:    vpblendvb %xmm4, %xmm0, %xmm2, %xmm0
-; AVX512-NEXT:    vpcmpeqw %xmm3, %xmm1, %xmm3
-; AVX512-NEXT:    vpblendvb %xmm3, %xmm1, %xmm0, %xmm0
-; AVX512-NEXT:    vcvtph2ps %xmm2, %ymm1
-; AVX512-NEXT:    vpxor %xmm3, %xmm3, %xmm3
-; AVX512-NEXT:    vcmpeqps %ymm3, %ymm1, %ymm1
-; AVX512-NEXT:    vpmovdw %zmm1, %ymm1
-; AVX512-NEXT:    vpblendvb %xmm1, %xmm0, %xmm2, %xmm0
-; AVX512-NEXT:    vzeroupper
+; AVX512-NEXT:    pushq %rbp
```
Comment (RKSimon) [inline llvm/test/CodeGen/X86/fminimum-fmaximum.ll:1644]: add nounwind to remove the cfi noise

File: llvm/test/CodeGen/X86/vector-reduce-fmax-nnan.ll [hunk 3]
```diff
@@ -413,9 +413,12 @@ define half @test_v2f16(<2 x half> %a0) nounwind {
...
 ; AVX512F-NEXT:    vpsrld $16, %xmm0, %xmm1
-; AVX512F-NEXT:    vcvtph2ps %xmm0, %ymm2
-; AVX512F-NEXT:    vcvtph2ps %xmm1, %ymm3
-; AVX512F-NEXT:    vcmpltps %zmm2, %zmm3, %k1
+; AVX512F-NEXT:    vcvtph2ps %xmm0, %xmm2
+; AVX512F-NEXT:    vcvtph2ps %xmm1, %xmm3
+; AVX512F-NEXT:    vucomiss %xmm3, %xmm2
+; AVX512F-NEXT:    seta %al
+; AVX512F-NEXT:    negb %al
+; AVX512F-NEXT:    kmovd %eax, %k1
```
Comment (RKSimon) [inline llvm/test/CodeGen/X86/vector-reduce-fmax-nnan.ll:421]: Annoying but not that important

**Review by RKSimon — APPROVED** [review]
LGTM with a couple of minors - cheers

File: llvm/test/CodeGen/X86/fminimum-fmaximum.ll (hunk 2 above)
Comment (phoebewang) [reply llvm/test/CodeGen/X86/fminimum-fmaximum.ll:1640]: Done.

File: llvm/test/CodeGen/X86/vector-reduce-fmax-nnan.ll (hunk 3 above)
Comment (phoebewang) [reply llvm/test/CodeGen/X86/vector-reduce-fmax-nnan.ll:421]: Yeah, but we don't need to tune much for unpractical combination.

**Pull request merged**